app.py

```diff
@@ -112,21 +111,90 @@ def get_bundles(entry):
     return bundles
 
 
-@app.route("/donate")
+def do_charge_or_show_errors(**kwargs):
+    app.logger.debug("----Retrieving Stripe customer...")
+
+    try:
+        email = request.form["stripeEmail"]
```

Comment:
I'd move everything out of the `try` block except the call to Stripe. It makes it easier to reason about and more obvious that everything after that won't be executed if there's an exception.

Reply:
Done.